src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -1,1058 +1,611 @@
+        var allFieldsBySObject = this.groupFieldsBySObject(component.get('v.everyField'));
+        var everyFieldUpdated = [];
+        Object.keys(allFieldsBySObject).forEach(function (currentSObject) {
+            var batchFieldGroups = component.get('v.availableFieldsBySObject').fieldGroups;
+            batchFieldGroups.forEach(function (currentFieldGroup) {
+                if (currentFieldGroup.sObjectName === currentSObject) {
+                    allFieldsBySObject[currentSObject].forEach(function (currentField) {
+                        currentField.isActive = currentFieldGroup.values.includes(currentField.id);
+                        // the field's sort order is its index PLUS the total of all active fields from all previous object groups
+                        currentField.sortOrder = currentField.isActive ? currentFieldGroup.values.indexOf(currentField.id) + fieldCountPreviousObjects : null;
+                        everyFieldUpdated.push(currentField);
+                    });
+                    // increase the buffer by the number of active fields from this object
+                    fieldCountPreviousObjects += currentFieldGroup.values.length;
+                }
+            });
+        });
+        component.set('v.everyField', everyFieldUpdated);
+    },
 
-            /**
-             * @description Updates the selected fields to Active, unselects fields
-             * @return void.
-             */
-            function updateBatchFieldOptions(batchFieldGroups) {
+    /**
+     * @description Updates batchFieldOptions attribute based on selected fields
+     */
+    updateBatchFieldOptions: function (component) {
+        let batchFieldOptions = {
+            fieldGroups: []
+        };
+        let activeFieldsBySObject = this.getActivesBySObject(component);
+        Object.keys(activeFieldsBySObject).forEach(function (sObjectName) {
 
-                var batchFieldOptions = [];
-                batchFieldGroups.forEach(function(currentFieldGroup) {
-                    currentFieldGroup.fields.forEach(function(currentField) {
-                        batchFieldOptions.push(currentField);
-                    });
-                });
+            var currentFieldGroup = {
+                sObjectName: sObjectName,
+                fields: []
+            };
 
-                _allFields.forEach(function(currentField) {
-                    batchFieldOptions.forEach(function(currentActiveField) {
-                        if (currentField.name === currentActiveField.name) {
-                            currentField.requiredInEntryForm = currentActiveField.requiredInEntryForm;
-                            currentField.hide = currentActiveField.hide;
-                            currentField.defaultValue = currentActiveField.defaultValue;
-                        }
-                    });
+            activeFieldsBySObject[sObjectName].forEach(function (currentField) {
+
+                var fieldInfo = {
```

Comment:
How is this different from pushing currentField into fields directly?

Reply:
great point. it looks like....it's not? this is likely leftover from the old pattern. i will test and report back...

Reply:
Yep, seems like we can just push currentField. Commit incoming.